File: apps/mybookkeeper/backend/app/schemas/insurance/insurance_policy_create_request.py
"""Schema for POST /insurance-policies."""
from __future__ import annotations

import datetime as _dt
import uuid

from pydantic import BaseModel, ConfigDict, Field


class InsurancePolicyCreateRequest(BaseModel):
    listing_id: uuid.UUID
    policy_name: str = Field(..., min_length=1, max_length=255)
    carrier: str | None = Field(None, max_length=255)
    policy_number: str | None = Field(None, max_length=255)
    effective_date: _dt.date | None = None
    expiration_date: _dt.date | None = None
    coverage_amount_cents: int | None = Field(None, ge=0)
    notes: str | None = Field(None, max_length=5000)

    model_config = ConfigDict(extra="forbid")

File: apps/mybookkeeper/frontend/src/shared/types/insurance/insurance-policy-create-request.ts
import type { InsurancePremiumFrequency } from "@/shared/types/insurance/insurance-premium-frequency";

/**
 * Request body for POST /insurance-policies.
 *
 * Mirrors ``schemas/insurance/insurance_policy_create_request.py``.
 */
export interface InsurancePolicyCreateRequest {
  listing_id: string;
  source_document_id?: string | null;
  policy_name: string;
  carrier?: string | null;
  policy_number?: string | null;
  effective_date?: string | null;
  expiration_date?: string | null;
  coverage_amount_cents?: number | null;
  premium_cents?: number | null;
  premium_frequency?: InsurancePremiumFrequency | null;
  deductible_cents?: number | null;
  wind_hail_deductible_pct?: string | null;
  notes?: string | null;
}
